### PR TITLE
lammps: add heffte support

### PR DESCRIPTION
**Pull request by junghans**

### CODE REVIEW

File: var/spack/repos/builtin/packages/lammps/package.py [hunk 1]
```diff
@@ -782,6 +787,8 @@ def url_for_version(self, version):
         msg="ROCm builds of the GPU package not maintained prior to version 20220623",
     )
     conflicts("+intel", when="%aocc@:3.2.9999", msg="+intel with AOCC requires version 4 or newer")
+    conflicts("+heffte", when="@:20240207", msg="heffte is only supported in 20240207 or newer")
```
Comment (rbberger) [inline var/spack/repos/builtin/packages/lammps/package.py]: That version range is inclusive. So you would also be preventing `20240207`.

File: var/spack/repos/builtin/packages/lammps/package.py [hunk 2]
```diff
@@ -782,6 +787,8 @@ def url_for_version(self, version):
...
+    conflicts("+heffte", when="~kspace", msg="heffte can only be used with kspace package")
```
Comment (rbberger) [inline var/spack/repos/builtin/packages/lammps/package.py]: might be worth making `heffte` a conditional variant then.

**Review by rbberger — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/lammps/package.py (hunk 1 above)
Comment (junghans) [reply var/spack/repos/builtin/packages/lammps/package.py]: Good catch, fixed!

File: var/spack/repos/builtin/packages/lammps/package.py (hunk 2 above)
Comment (junghans) [reply var/spack/repos/builtin/packages/lammps/package.py]: I didn't know this was possible now, so changed.

**Comment by junghans** [comment]
@spackbot fix style

**Review by mkstoyanov — APPROVED** [review]

**Review by rbberger — APPROVED** [review]

**Comment by junghans** [comment]
@spack/merge-to-develop, please merge!

**Review by tldahlgren — APPROVED** [review]
Confirmed the version sha256.

**Pull request merged**